x-pack/plugins/actions/server/sub_action_framework/sub_action_connector.ts

```diff
@@ -140,6 +142,19 @@ export abstract class SubActionConnector<Config, Secrets> {
           `Request to external service failed. Connector Id: ${this.connector.id}. Connector type: ${this.connector.type}. Method: ${error.config.method}. URL: ${error.config.url}`
         );
 
+        let responseBody = '';
+
+        // The error response body may also be a stream, e.g. for the GenAI connector
+        if (error.response?.config.responseType === 'stream' && error.response?.data) {
+          const incomingMessage = error.response.data as IncomingMessage;
+
+          incomingMessage.on('data', (chunk) => {
+            responseBody += chunk.toString();
+          });
+          await finished(incomingMessage);
+          error.response.data = JSON.parse(responseBody);
```

Comment:
This seems ok, generally.  But a couple of things:

- wrap the `JSON.parse()` with a try catch, to catch cases where it's not JSON for some reason
- add a jest test
- add a function test, assuming we have some decent function tests for this

Reply:
I've wrapped it in try/catch and have added an API test in [`27b7c7a` (#162908)](https://github.com/elastic/kibana/pull/162908/commits/27b7c7a0cc01e9842661f376f6823a5a3a470502). I could not find existing API tests, other than the ones for Observability AI Assistant, as we're the first consumer of this option. So I've added one there. I feel like a unit test here is less useful because there's so much stuff to mock out.

Reply:
Unit tests are generally useful to spot problems earlier and more precisely than functional tests, but understand with all the mocking :-).  I susect that if this code causes some issues in the future, we can add some jest tests then.